drivers/platform/stm32/stm32_dma.c

```diff
@@ -60,6 +60,7 @@ static struct no_os_dma_desc *dma_descriptor;
 int stm32_dma_config_xfer(struct no_os_dma_ch *channel,
 			  struct no_os_dma_xfer_desc *xfer)
 {
+#ifdef HAL_DMA_MODULE_ENABLED
```

Comment:
why would you use stm32_dma.c if HAL_DMA_MODULE_ENABLED is not defined? I think this check is not needed...
only use this check in stm32_spi, because the spi driver can be used without the DMA module.
also, this change is not related to this commit and should be part of the first commit.